### PR TITLE
Summary: Add Binary Index i.o to c_api

### PR DESCRIPTION
**Pull request by CarterMcClellan**
Summary:
Add Binary Index i.o to c_api

In reference to isse: https://github.com/facebookresearch/faiss/issues/2225

### CODE REVIEW

**Comment by mdouze** [comment]
@Enet4 any thoughts about this diff?

File: c_api/IndexBinary_c.h [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+extern "C" {
+#endif
+
+// forward declaration required here
+FAISS_DECLARE_CLASS(RangeSearchResult)
+
+// typedef struct FaissRangeSearchResult_H FaissRangeSearchResult;
+typedef struct FaissIDSelector_H FaissIDSelector;
+
+// /// Opaque type for referencing to an index object
```
Comment (Enet4) [inline c_api/IndexBinary_c.h:28]:
```suggestion
/// Opaque type for referencing to a binary index object
```

**Review by Enet4 — COMMENTED** [review]
I have no objections to this, it seems to have followed the pattern well. :+1:

Only found a nitpicky typo, but merging would be fine by me once that's fixed.

File: c_api/IndexBinary_c.h (hunk 1 above)
Comment (CarterMcClellan) [reply c_api/IndexBinary_c.h]: :+1: comment should now be fixed

**Review by Enet4 — APPROVED** [review]

**Pull request closed without merging**